packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dq-response.ts

```diff
@@ -186,21 +184,27 @@ function handleSuccessResponse({
     gateway,
     documentReference: documentReferences,
     externalGatewayPatient: outboundRequest.externalGatewayPatient,
+    retried: attempt,
   };
   return response;
 }
 
-export function processDQResponse({
-  dqResponse: { response, success, gateway, outboundRequest },
+export function processDqResponse({
+  response: { response, success, gateway, outboundRequest },
+  attempt,
 }: {
-  dqResponse: DQSamlClientResponse;
-}): OutboundDocumentQueryResp {
+  response: DQSamlClientResponse;
+  attempt?: number | undefined;
+}): Promise<OutboundDocumentQueryResp> {
   if (success === false) {
-    return handleHttpErrorResponse({
-      httpError: response,
-      outboundRequest,
-      gateway: gateway,
-    });
+    return Promise.resolve(
```

Comment:
make this return a promise so that the return types of processDqResponse and processDrResponse are both promises and can work with the same exponential backoff wrapper function

Reply:
Just make the function `async`, then we don't need to wrap it around a promise.

Here and elsewhere

Reply:
I see you 👍 but the code still resolves a promise, when we could just return `handleHttpErrorResponse` if we add `async` to the function signature, right?

Reply:
sorry I missed that one.